docs/architecture/011-capi-elements-as-components.md

```diff
@@ -0,0 +1,27 @@
+# Static types
+
+## Context
+
+CAPI provides a representation of content which is strongly typed and like components.
+
+### CAPI Elements as Components
+
+The CAPI Blocks API exposes a suite of strongly typed elements. A component should be made for each element to render it.
+
+### Functionality
```

Comment:
My instinct is to try and maintain the separation between rendering and CAPI that frontend currently provides on the assumption that CAPI may not always suit what is needed here so a separate model, which rendering defines is probably useful.

But some examples of what it might look like would help here.

Reply:
Examples would be good to illustrate this as I'm not familiar enough with block structure, but I like the idea of mapping a strongly typed block-type from CAPI directly to a Component in the rendering layer. I do agree with Nic's point that frontend should  provide data in a spec defined by the rendering layer. Is there anywhere we can see a list of supported 'block types' from CAPI?

Reply:
This is the block structure from 
![image](https://user-images.githubusercontent.com/2670496/46796331-51e17380-cd44-11e8-8b74-11d387579126.png) https://www.theguardian.com/food/2018/oct/11/kombucha-can-the-fermented-drink-compete-with-beer-at-the-bar.json?guui

Reply:
It feels like the frontend repo should be defining the block model expected by dotcom-rendering. For instance, it might be necessary for us to render a component that is not defined by a data structure not present in CAPI, but is instead entirely invented by frontend. In that respect, is CAPI even worth mentioning here?

Reply:
NB: These contain multiple `<p>...</p>`s, it's just firefox squashing it up.

This document should link to whereever we discuss what happens to CAPI data to renderable. The code might not need to know, but I think the docs should.